current/standard-derivatives.md

```diff
@@ -0,0 +1,552 @@
+# Standard Derivatives
+
+**How do I...**
+
+- [Create the Archives Unleashed Cloud Scholarly Derivatives](#Create-the-Archives-Unleashed-Scholarly-Derivatives)
+- [Extract Binary Info to CSV](#Extract-Binary-Info-to-CSV)
+- [Extract Binary Info to CSV (s3)](#Extract-Binary-Info-to-CSV-s3)
+- [Extract Binary Info to Parquet](#Extract-Binary-Info-to-Parquet)
+- [Extract Binaries to Disk](#Extract-Binaries-to-Disk)
+
+For all the scripts below, you can type `:paste` into Spark Shell, paste the script, and then run it with <kbd>CTRL</kbd>+<kbd>d</kbd>:
+
+## Create the Archives Unleashed Cloud Scholarly Derivatives
+
+How do I create the [scholarly derivatives](https://cloud.archivesunleashed.org/derivatives) that the Archives Unleashed Cloud creates on my own web archive collection?
+
+Note, the full-text and domains output needs to be concatenated together into a single file respectively to replicate the Cloud output, and the GraphML file needs to be run through [`graphpass`](https://github.com/archivesunleashed/graphpass) with the following command:
+
+```bash
+$ graphpass input.graphml output.gexf -gq
+```
+
+### Scala RDD
+
+```scala
+import io.archivesunleashed._
+import io.archivesunleashed.app._
+import io.archivesunleashed.matchbox._
+
+sc.setLogLevel("INFO")
+
+// Web archive collection.
+warcs = RecordLoader
+  .loadArchives("/path/to/data", sc)
+  .keepValidPages()
+
+// Domains file.
+warcs
+  .map(r => ExtractDomain(r.getUrl))
+  .countItems()
+  .saveAsTextFile("/path/to/derivatives/auk/all-domains/output")
+
+// Full-text.
+warcs
+  .map(r => (r.getCrawlDate, r.getDomain, r.getUrl, RemoveHTML(RemoveHttpHeader(r.getContentString))))
+  .saveAsTextFile("/path/to/derivatives/auk/full-text/output")
+
+// Gephi GraphML.
+val links = warcs
+  .map(r => (r.getCrawlDate, ExtractLinks(r.getUrl, r.getContentString)))
+  .flatMap(r => r._2.map(f => (r._1, ExtractDomain(f._1).replaceAll("^\\\\s*www\\\\.", ""), ExtractDomain(f._2).replaceAll("^\\\\s*www\\\\.", ""))))
+  .filter(r => r._2 != "" && r._3 != "")
+  .countItems()
+  .filter(r => r._2 > 5)
+
+WriteGraph
+  .asGraphml(links, "/path/to/derivatives/auk/graph/example.graphml")
+
+sys.exit
+```
+
+### Scala DF
+
+TODO
+
+### Python DF
+
+TODO
+
+## Extract Binary Info to CSV
+
+How do I extract binary information of PDFs, audio files, video files, word processor files, spreadsheet files, presentation program files, and text files to a CSV file?
+
+### Scala RDD
+
+TODO
+
+### Scala DF
+
+```scala
+import io.archivesunleashed._
+import io.archivesunleashed.df._
+
+sc.setLogLevel("INFO")
+
+// Web archive collection.
+val warcs = RecordLoader.loadArchives("/path/to/data", sc)
+
+// Audio Files.
+warcs
+  .audio()
+    .select($"url", $"filename", $"extension", $"mime_type_web_server", $"mime_type_tika", $"md5")
```

Comment:
I think this should be the same indent level?

```
warcs
  .audio()
  .select($"url", $"filename", $"extension", $"mime_type_web_server", $"mime_type_tika", $"md5")
```
Although I think this looks more natural:

```
warcs.audio()
  .select($"url", $"filename", $"extension", $"mime_type_web_server", $"mime_type_tika", $"md5")
```